src/components/accessories/autocompleteField/AutocompleteField.tsx

```diff
@@ -0,0 +1,65 @@
+import { FormControl, FormHelperText } from "@material-ui/core";
+import TextField from "@material-ui/core/TextField";
+import React, {
+  Fragment,
+  FunctionComponent,
+  memo,
+  useEffect,
+  useState,
+} from "react";
+import { IProps } from "./types";
+import "./styles.scss";
+import { Autocomplete } from "@material-ui/lab";
+
+const AutocompleteField: FunctionComponent<IProps> = ({
+  fieldName,
+  fieldValue,
+  label,
+  isValid,
+  errorText,
+  onBlur,
+  options,
+}) => {
+  const [value, setValue] = useState("");
+
+  const geFullObj = (val: string) => {
+    return options?.find((el) => el.value === val) || null;
+  };
+  const handleOnBlur = (e: React.FocusEvent<HTMLDivElement>) => {
+    onBlur(e, value);
+  };
+
+  useEffect(() => {
+    setValue(fieldValue);
+  }, [fieldValue]);
+
+  return (
+    <FormControl variant="outlined" className="autocomplete" size="small">
+      <Autocomplete
+        freeSolo
+        options={options}
+        getOptionLabel={(option) => option.label}
+        value={geFullObj(value)}
+        getOptionSelected={(option, v) => option.value === v.value}
+        renderOption={(option) => <Fragment>{option.label}</Fragment>}
+        onChange={(e: object, val: any | null) => {
+          setValue(val?.value || "");
+        }}
```

Comment:
We should add a debounce function in order to prevent the status updated every single time a user types on a key.

Something like that:

`onChange={(e: object, val: any | null) => {
  debounceUpdate(val?.value || "");
}}

const debounceUpdate= useCallback(
  debounce(
    (value) =>
      setValue(value),
    250
  ),
  []
);
`

Reply:
I'm not sure if the useCallback couldn't cause some throubles, if it does, try to remove it.